src/python/pants/backend/go/goals/test.py

```diff
@@ -189,31 +200,41 @@ def compilation_failure(exit_code: int, stdout: str | None, stderr: str | None)
     testmain = await Get(
         GeneratedTestMain,
         GenerateTestMainRequest(
-            pkg_digest.digest,
-            FrozenOrderedSet(
+            digest=pkg_digest.digest,
+            test_paths=FrozenOrderedSet(
                 os.path.join(".", pkg_analysis.dir_path, name)
                 for name in pkg_analysis.test_go_files
             ),
-            FrozenOrderedSet(
+            xtest_paths=FrozenOrderedSet(
                 os.path.join(".", pkg_analysis.dir_path, name)
                 for name in pkg_analysis.xtest_go_files
             ),
-            import_path,
-            field_set.address,
+            import_path=import_path,
+            register_cover=test_subsystem.use_coverage,
+            address=field_set.address,
         ),
     )
 
+    _logger.info(f"testmain = {testmain}")
```

Comment:
TODO

While at it: the convention elsewhere is to name global loggers just `logger`, unprefixed.